examples/eSealing/src/lib.rs

```diff
@@ -0,0 +1,328 @@
+//! A contract implementing an eSealing service.
+//!
+//! If you seal a file, you can prove that it was in your possession at the time
+//! of sealing. The dApp flow will be:
+//! - Upload a file from the computer at the front-end => register its file hash
+//!   in the smart contract
+//! - Upload a file from the computer at the front-end => retrieve the timestamp
+//!   and witness (sender_account) from the smart contract to prove that the
+//!   witness (sender_account) was in possession of that file at that time.
+//!
+//! Only accounts can register a file hash. During the registration, the
+//! timestamp is recorded together with the witness (sender_account) that
+//! registered the file hash. Each file hash can only be registered once.
+//! This is because we want to record the first time when the witness
+//! (sender_account) had access to that file. Re-registering a file hash (by a
+//! different witness) would not prove that the second witness is also in
+//! possession of that file because the second witness could have read the
+//! file hash during the initial registration transaction from the blockchain.
+use concordium_std::*;
+
+/// The different errors the contract can produce.
+#[derive(Serialize, Debug, PartialEq, Eq, Reject, SchemaType)]
+enum ContractError {
+    /// Failed parsing the parameter.
+    #[from(ParseError)]
+    ParseParams,
+    /// Failed logging: Log is full.
+    LogFull,
+    /// Failed logging: Log is malformed.
+    LogMalformed,
+    /// Only accounts can register a file hash.
+    OnlyAccount,
+    /// Each file hash can only be registered once.
+    AlreadyRegistered,
+}
+
+/// Mapping the logging errors to ContractError.
+impl From<LogError> for ContractError {
+    fn from(le: LogError) -> Self {
+        match le {
+            LogError::Full => Self::LogFull,
+            LogError::Malformed => Self::LogMalformed,
+        }
+    }
+}
+
+/// The state tracked for each file.
+#[derive(Serial, Deserial, Clone, Copy, SchemaType)]
+struct FileState {
+    /// The timestamp when this file hash was registered.
+    timestamp: Timestamp,
+    /// The witness (sender_account) that registered this file hash.
+    witness:   AccountAddress,
+}
+
+/// FileHash type is a fixed-length byte array.
+type FileHash = [u8; 32];
```

Comment:
This does allow you to specify the hash as a hex string in JSON, right?

Reply:
I don't think it does. It needs the `ByteArray(32)` schema type for this and I think this will have the more generic `Array(32, U8)`.

Reply:
The front-end/concordium-client require as input parameter a `ByteArray(32)` in the format:
`[1,2, ..,32]`

a hex string does not work

Reply:
Well could we make it work then please? That will be a better experience. I know this is merged now, but can we please can you fix this?